### ISSUE TITLE
Folding 'List' constructor with initial capacity is misleading

### ISSUE THREAD

**Issue by sabi0** [cheptsov/AdvancedExpressionFolding#105]
![image](https://user-images.githubusercontent.com/11509875/35156182-5a309db6-fd30-11e7-9b7a-ef5b1292f8c4.png)

Maybe it should be folded to something like `hCharSequence = [4];` instead?

**Comment by ciscorucinski** [comment]
I believe that people use the `initialCapacity` because they care about performance. Therefore, the folding should be similar but different enough to indicate a different usage

My initial idea is one of the following 3

    ArrayList<String> list_4 = new ArrayList<>(4);
    var list_4 = [ , , , ]    // 1.
    var list_4 = [_ _ _ _]    // 2. Or this
    var list_4 = [...]        // 3. Or this


    ArrayList<String> list_50 = new ArrayList<>(50);
    var list_50 = [ , , ... , ]   // 1.
    var list_50 = [_ _ ... _]     // 2. Or this
    var list_50 = [...]           // 3. Or this

As far as the fold is concerned, all a coder cares about is that it IS using the `initialCapacity` constructor. 

Seeing the number is not that important, and that number isn't changed much after it is declared. If I need to see the number, I can hover and see if that is correct. Therefore, I believe one of the above 3 ideas is sufficient.

What do you think @sabi0 and @cheptsov ? Any other ideas?

**Comment by sabi0** [comment]
I agree, `[4]` is confusing too. I hesitated about suggesting it. But didn't come up with a better option.
I am not very much in favor of hiding the value. Maybe something like `[] x 4`?
